src/main/kotlin/com/michaelsgroi/baseballreference/CachedFile.kt

```diff
@@ -0,0 +1,54 @@
+package com.michaelsgroi.baseballreference
+
+import java.io.File
+import java.time.Duration
+import java.time.Instant
+
+class CachedFile {
+    companion object {
+        fun loadFromCache(filename: String, expiration: Duration, loader: () -> String): List<String> {
+            if (!filename.fileExists()) {
+                println("filename=$filename does not exist, retrieving from baseball-reference.com ...")
+                filename.writeFile(loader())
+            } else {
+                if (filename.fileExpired(expiration)) {
+                    println("filename=$filename exists but is expired, retrieving from baseball-reference.com ...")
+                    filename.writeFile(loader())
+                }
+            }
+            return filename.readFile()
+        }
+    }
+}
+
+fun String.readFile(): List<String> {
```

Comment:
why not have these be extensions on `File` (instead of `String`) (one reason: avoid creating new `File` objects; also, then `File` objects will be passed around, whose meaning/purpose (i.e., representing a file) is likely clearer than a `String` object)

Reply:
yeah, this is the kind of thing I would never do in someone else's code but I've realized is just so brutally simple for me to deal with personally that I end up copying it to every project

Reply:
seems like it'd be actually _easier_, especially in these methods, with `File` (and only a smidge more code for callers—but also easier to understand on that end)... definitely opinion though